scripts/release.js

```diff
@@ -0,0 +1,214 @@
+// nothing is pushed nor published remove local changes
+function rollback(newVersion) {
+  if (strategy === 'stable') {
+    // reset master
+    execSync('git reset --hard origin master');
+    execSync('git checkout develop');
+  } else {
+    // remove last commit
+    execSync('git reset --hard HEAD~1');
+  }
+
+  // remove local created tag
+  execSync(`git tag -d v${newVersion}`);
+  process.exit(1);
+}
+
+inquirer
+  .prompt([
+    {
+      type: 'confirm',
+      name: 'confirm',
+      message: colors.yellow.underline(
+        `You are on "${currentBranch}" branch. Are you sure you want to release a ${strategy ===
+        'stable'
+          ? 'new stable'
+          : 'beta'} version?`
+      ),
+      default: false,
+    },
+  ])
+  .then(({ confirm }) => {
+    if (!confirm) process.exit(0);
+
+    if (strategy === 'stable') {
+      // merge develop on master first
+      // and start release from master branch
+      console.log(colors.blue('Updating working tree'));
+      execSync('git checkout master');
+      execSync('git pull origin master');
+      execSync('git fetch origin --tags');
+      execSync('git fetch origin develop');
+      execSync('git merge origin/develop');
+    }
+
+    console.log(colors.blue('Install dependencies'));
+    execSync('yarn');
+
+    const { version: currentVersion } = require('../package.json');
+    console.log(
+      colors.blue(`
+      - Current version is "${currentVersion}"
+      - Changelog will be generated only if a fix/feat/performance/breaking is found in git log
+      - You must choose a new ve.rs.ion (semver) ${strategy === 'beta'
+        ? 'with -beta.x suffix'
+        : ''}
+    `)
+    );
+
+    execSync(
+      'conventional-changelog --preset angular --output-unreleased | less'
+    );
+
+    inquirer
+      .prompt([
+        {
+          type: 'input',
+          name: 'newVersion',
+          message: colors.blue.underline(
+            strategy === 'stable'
+              ? 'Please type the new chosen version'
+              : 'Please type the new chosen version (with -beta.x suffix)'
+          ),
+          validate(newVersion) {
+            return (
+              strategy === 'stable' ||
+              (newVersion && /-beta\.\d{1,}$/.test(newVersion)) ||
+              'You must include `-beta.x` suffix in the version number'
+            );
+          },
+        },
+      ])
+      .then(({ newVersion }) => {
+        // bump new version
+        console.log(colors.blue(`Bump version to "${newVersion}"`));
+        execSync(
+          `VERSION=${newVersion} babel-node ./scripts/bump-package-version.js`
+        );
+
+        // build library new version
+        console.log(colors.blue('Building library new version'));
+        execSync(`NODE_ENV=production VERSION=${newVersion} npm run build`);
+
+        // update changelog
+        console.log(colors.blue('Update changelog'));
+        const changelog = execSync('conventional-changelog -p angular')
```

Comment:
Because we need the changelog for the commit message and I think that a changelog for beta versions could be good as well. Maybe reset it before a new stable version, wdyt?